packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/contexts/NodeContext.types.ts

```diff
@@ -0,0 +1,35 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { NodeProps } from 'reactflow';
```

Comment:
- The file is named `NodeContext` but the context is not
- Why this registry with functions (I know why because it is only the latest copy / paste from my first example of an extension of the frontend which was a very basic registry a couple years ago) but it does not have to be this complex. Something like this should work just as efficiently:

```
export interface NodeTypeContextValue {
  nodeStyleContributions: NodeStyleContribution[];
  nodeLayoutHandlers: INodeLayoutHandler<NodeData>[];
  nodeConverters: INodeConverter[];
  nodeTypeContributions: NodeTypeContributionElement[]
}
```

No need for a registry, no need for functions, just data or am I mistaken?

Reply:
Done